x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutTrainedModelDefinitionPartAction.java

```diff
@@ -57,12 +55,12 @@ public static Request parseRequest(String modelId, int part, XContentParser pars
         }
 
         private final String modelId;
-        private final BytesReference definition;
+        private final String definition;
```

Comment:
The reason for making this change is not clear to me. We expect the definition to be base64 encoded, Java UTF-16 characters require 2 bytes to store each base64 character whereas the raw bytes (given UFT-8 encoding) will use half that space.

Reply:
Reason for the change is to do `definition_length` we need to decode the base64 values. To do that, it needs to be a string. So, might as well keep it as a string

Reply:
It isn't a big deal, I can switch it back, I will just have to transform it to a string when decoding it.